hs_core/tasks.py

```diff
@@ -2,45 +2,53 @@
 
 from __future__ import absolute_import
 
+import logging
 import os
 import sys
 import traceback
 import zipfile
-import logging
-
-import requests
-
+from datetime import timedelta, date
 from xml.etree import ElementTree
 
-from rest_framework import status
-
+import requests
+from celery import shared_task
+from celery.schedules import crontab
+from celery.task import periodic_task
 from django.conf import settings
-from django.core.mail import send_mail
 from django.contrib.auth.models import User
+from django.core.mail import send_mail
+from rest_framework import status
 
-from celery.task import periodic_task
-from celery.schedules import crontab
-from celery import shared_task
-
-from hs_core.models import BaseResource
+from django_irods.icommands import SessionException
+from django_irods.storage import IrodsStorage
 from hs_core.hydroshare import utils
 from hs_core.hydroshare.hs_bagit import create_bag_files
 from hs_core.hydroshare.resource import get_activated_doi, get_resource_doi, \
     get_crossref_url, deposit_res_metadata_with_crossref
-from django_irods.storage import IrodsStorage
+from hs_core.models import BaseResource
 from theme.models import UserQuota, QuotaMessage
 from theme.utils import get_quota_message
 
-from django_irods.icommands import SessionException
-
-
 # Pass 'django' into getLogger instead of __name__
 # for celery tasks (as this seems to be the
 # only way to successfully log in code executed
 # by celery, despite our catch-all handler).
 logger = logging.getLogger('django')
 
 
+@periodic_task(ignore_result=True, run_every=crontab(minute=30, hour=23))
+def nightly_zips_cleanup():
+    # delete 2 days ago
+    date_folder = (date.today() - timedelta(2)).strftime('%Y-%m-%d')
+    zips_daily_date = "zips/{daily_date}".format(daily_date=date_folder)
+    istorage = IrodsStorage()
+    if istorage.exists(zips_daily_date):
+        istorage.delete(zips_daily_date)
+        logger.debug("Removed daily zip folder " + zips_daily_date)
+    else:
+        logger.debug("No daily date folder found for " + zips_daily_date)

```

Comment:
@sblack-usu I'd suggest to remove these debug logs once you made sure the code works since this task is just a precaution in case there were leftover temp zip folders. Otherwise, these debug messages will go into our logs each day which will clutter our log.

Reply:
Are we logging at the debug level in production?  I would think we'd be at the info level, so this shouldn't be written out daily.  I thought it'd be a good idea to have the ability to write something to logs in case of the need to debug.

Reply:
@sblack-usu I think we are logging at the debug level in production. @aphelionz Can you verify?

Reply:
I'm willing to remove these if necessary, but if we're regularly logging out at debug level, we may want to evaluate our logging practices

Reply:
@sblack-usu I'd argue that when there were an issue related to zipping folders, you can always put in this debug log for diagnosis. In this specific case, you can always check on iRODS side to see if there are any left over zip folders to see whether there were issues with cleanup or not.

Reply:
I've always used debug level logging for messages that are helpful when debugging an issue, something you can turn on as needed without pushing a code change.  I'll go ahead and remove them so I can get this merged as I'm not concerned about these logs in particular.